lib/ASTGen/Sources/ASTGen/Generics.swift

```diff
@@ -87,4 +89,117 @@ extension ASTGenVisitor {
+          fatalError("Unknown prefix operator for same type left type")
+        }
+
+        guard case .integerLiteralExpr(let node) = node.expression.as(ExprSyntaxEnum.self) else {
+          fatalError("Unknown expression kind for same type left type")
+        }
+
+        return self.generate(integerType: node, minusLoc: self.generateSourceLoc(op))
+
+      default:
+        fatalError("Unknown expression kind for same type left type")
+      }
+    }
+  }
+
+  func generate(sameTypeRightType node: SameTypeRequirementSyntax.RightType) -> BridgedTypeRepr {
+    switch node {
+    case .type(let type):
+      return self.generate(type: type)
+
+    // The only expressions same type right types support right now are
+    // integer literals, '123', and prefix operators for negative integer
+    // literals, '-123'.
+    case .expr(let expr):
+      switch expr.as(ExprSyntaxEnum.self) {
+      case .integerLiteralExpr(let node):
+        return self.generate(integerType: node)
+
+      case .prefixOperatorExpr(let node):
+        let op = node.operator
+
+        guard op.text == "-" else {
+          fatalError("Unknown prefix operator for same type right type")
+        }
+
+        guard case .integerLiteralExpr(let node) = node.expression.as(ExprSyntaxEnum.self) else {
+          fatalError("Unknown expression kind for same type right type")
+        }
+
+        return self.generate(integerType: node, minusLoc: self.generateSourceLoc(op))
+
+      default:
+        fatalError("Unknown expression kind for same type right type")
+      }
+    }
+  }
+
+  func generate(genericArgument node: GenericArgumentSyntax.Argument) -> BridgedTypeRepr {
+    switch node {
+    case .type(let type):
+      return self.generate(type: type)
+
+    // The only expressions generic argument types support right now are
+    // integer literals, '123', and prefix operators for negative integer
+    // literals, '-123'.
+    case .expr(let expr):
+      switch expr.as(ExprSyntaxEnum.self) {
+      case .integerLiteralExpr(let node):
+        return self.generate(integerType: node)
+
+      case .prefixOperatorExpr(let node):
+        let op = node.operator
+
+        guard op.text == "-" else {
+          fatalError("Unknown prefix operator for generic argument type")
+        }
+
+        guard case .integerLiteralExpr(let node) = node.expression.as(ExprSyntaxEnum.self) else {
+          fatalError("Unknown expression kind for generic argument type")
+        }
+
+        return self.generate(integerType: node, minusLoc: self.generateSourceLoc(op))
+
+      default:
+        fatalError("Unknown expression kind for generic argument type")
+      }
+    }
+  }
+
+  func generate(
+    integerType node: IntegerLiteralExprSyntax,
+    minusLoc: BridgedSourceLoc = BridgedSourceLoc()
+  ) -> BridgedTypeRepr {
+    var desc = node.trimmedDescription
+    let str = desc.withBridgedString {
+      self.ctx.allocateCopy(string: $0)
+    }
```

Comment:
Could you use `self.copyAndStripUnderscores(text: node.rawText)` (where `node` is `IntegerLiteralExprSyntax`)